cf/data/data.py

```diff
@@ -3831,304 +3831,70 @@ def _binary_operation(self, other, method):
-                align_offset = -align_offset
-                shape0 = data0_shape
-                shape1 = data1_shape[align_offset:]
-
-                new_shape = data1_shape[:align_offset]
-                new_ndim = data1_ndim
-                if not data0_ndim:
-                    new_axes = other._axes
-                else:
-                    new_axes = []
-                    existing_axes = self._all_axis_names()
-                    for n in new_shape:
-                        axis = new_axis_identifier(existing_axes)
-                        existing_axes.append(axis)
-                        new_axes.append(axis)
-                    # --- End: for
-                    new_axes += data0._axes
-                # --- End: for
-
-                align_offset = 0
-            # --- End: if
-
-            broadcast_indices = []
-            for a, b in zip(shape0, shape1):
-                if a == b:
-                    new_shape += (a,)
-                    broadcast_indices.append(None)
-                    continue
-
-                # Still here?
-                if a > 1 and b == 1:
-                    new_shape += (a,)
-                elif b > 1 and a == 1:
-                    new_shape += (b,)
-                else:
-                    raise ValueError(
-                        "Can't broadcast shape {} against shape {}".format(
-                            data1_shape, data0_shape
-                        )
-                    )
-
-                broadcast_indices.append(slice(None))
-
-            new_size = reduce(mul, new_shape, 1)
-
-            dummy_location = [None] * new_ndim
-        # ---End: if
-
-        new_flip = []
+        data0 = self.copy()
 
-        # ------------------------------------------------------------
-        # Create a Data object which just contains the metadata for
-        # the result. If we're doing a binary arithmetic operation
-        # then result will get filled with data and returned. If we're
-        # an augmented arithmetic assignment then we'll update self
-        # with this new metadata.
-        # ------------------------------------------------------------
+        # Parse units
+        data0, other, new_Units = data0._combined_units(other, method, True)
 
-        result = data0.copy()
-        result._shape = new_shape
-        result._ndim = new_ndim
-        result._size = new_size
-        result._axes = new_axes
+        # Cast as dask arrays
+        dx0 = data0._get_dask()
+        dx1 = other._get_dask()
 
-        # ------------------------------------------------------------
-        # Set the data-type of the result
-        # ------------------------------------------------------------
-        if method_type in ("_eq", "_ne", "_lt", "_le", "_gt", "_ge"):
-            new_dtype = np.dtype(bool)
+        # Set if applicable the tolerance levels for the result
+        if method in ("__eq__", "__ne__"):  # what about l/g-t-e (x4)?
```

Comment:
What does `l/g-t-e (x4)` mean?

Reply:
Having consulted the cryptonomicon, I see now!

Reply:
Sorry, that was my very arcane way to try to convery `lt`, `gt`, `lt` and `gt` in slightly less characters :grimacing: In the original code we are assigning `rtol` and `atol` values for all of those:

https://github.com/NCAS-CMS/cf-python/blob/032eabd4ca2f56c7b52464e4eec89252cc48df3e/cf/data/data.py#L3516-L3519

but then we don't seem to use it in the inequality cases, so I don't think it is required there despite those methods being listed in the statement? Just to check I am not missing something here?

Reply:
I recalling working out that tolerant lt, le, ge, gt comparisons were not a good idea/didn't make sense. I guess setting rtol and atol in that if statement was just convenient.